src/ITensorTDVP.jl

```diff
@@ -1,19 +1,9 @@
 module ITensorTDVP
-export TimeDependentSum, dmrg_x, linsolve, tdvp, to_vec
 
-include("defaults.jl")
-include("update_observer.jl")
-include("solver_utils.jl")
-include("tdvporder.jl")
-include("tdvpinfo.jl")
-include("sweep_update.jl")
-include("alternating_update.jl")
-include("tdvp.jl")
-include("dmrg.jl")
-include("dmrg_x.jl")
-include("projmpo_apply.jl")
-include("contract_mpo_mps.jl")
-include("projmps2.jl")
-include("projmpo_mps2.jl")
-include("linsolve.jl")
+using Reexport: @reexport
+@reexport using ITensors.ITensorMPS: tdvp, dmrg_x, to_vec, TimeDependentSum, linsolve
+
+using ITensors.ITensorMPS: ITensorMPS
+dmrg(args...; kwargs...) = ITensorMPS.alternating_update_dmrg(args...; kwargs...)
```

Comment:
This kind of thing gives me pause and I want to make sure we think this through properly.

Is `ITensorMPS.alternating_update_dmrg` meant to be an advanced version of `dmrg` based on `alternating_update` with a forward-looking interface?

If so, defining `ITensorTDVP.dmrg` directly based on it is not a good idea, since we will very likely be changing `ITensorMPS.alternating_update_dmrg` and therefore cause a breaking change to this function.

We probably want a buffer function defined in `ITensors.ITensorMPS`, say called `itensortdvp_dmrg`, which wraps `alternating_update_dmrg`, and tries to keep the interface of the current `ITensorTDVP.dmrg`. The same kind of thing could go for the other solvers, though those are a bit more subtle since some of them overload `KrylovKit` functions.

Reply:
Yes I see what you mean about the 'extra layer of indirection' here. My definition of this was just a pretty literal "port" of the ITensorTDVP.dmrg function, so definining it exactly the same way it was previously defined in this package just changing the name. 

We could think of `alternating_update_dmrg` in fact as the backwards-compatibility layer you are proposing, which is closer what I intended it as, though in that case I agree that a name like `itensortdvp_dmrg` is better to make it explicit that it's really about supporting a certain package at a certain version. I'll make that change and others along the lines discussed over Slack.